internal/host/validator.go

```diff
@@ -342,6 +343,49 @@ func (v *validator) printCompatibleWithClusterPlatform(c *validationContext, sta
 	}
 }
 
+func isDiskEncryptionRelevantForHost(c *validationContext) bool {
+	switch *c.cluster.DiskEncryption.EnableOn {
+	case models.DiskEncryptionEnableOnAll:
+		return true
+	case models.DiskEncryptionEnableOnMasters:
+		return c.host.Role == models.HostRoleMaster || c.host.Role == models.HostRoleBootstrap
+	case models.DiskEncryptionEnableOnWorkers:
+		return c.host.Role == models.HostRoleWorker
```

Comment:
What do we do with auto-assign?  I recommend assuming that if it's enabled on masters or workers, then it will be enabled on hosts that are auto-assigned.  @slaviered is working on knowing the role earlier on so we can fix this later.

Reply:
Not sure how your suggestion works, seems to me like we can enable disk-encryption on auto-assigned hosts only if `enable_on == 'all'`

Reply:
Maybe the behavior should be that if at least one host is auto-assign then the user must choose none/all?

Reply:
Sound like a good and easy solution to me.

Reply:
Fixed.